lib/Frontend/SerializedDiagnosticConsumer.cpp

```diff
@@ -154,25 +154,41 @@ class SerializedDiagnosticConsumer : public DiagnosticConsumer {
       exitDiagBlock();
 
     // Write the generated bitstream to the file.
-    std::error_code EC;
-    std::unique_ptr<llvm::raw_fd_ostream> OS;
-    OS.reset(new llvm::raw_fd_ostream(State->SerializedDiagnosticsPath, EC,
-                                      llvm::sys::fs::F_None));
-    if (EC) {
-      // Create a temporary diagnostics engine to print the error to stderr.
-      DiagnosticEngine DE(SM);
-      PrintingDiagnosticConsumer PDC;
-      DE.addConsumer(PDC);
-      DE.diagnose(SourceLoc(), diag::cannot_open_serialized_file,
-                  State->SerializedDiagnosticsPath, EC.message());
+    std::unique_ptr<llvm::raw_fd_ostream> OS = openOutputFile();
+    if (!OS)
       return true;
-    }
 
     OS->write((char *)&State->Buffer.front(), State->Buffer.size());
     OS->flush();
     return false;
   }
 
+  bool informDriverThatCompilationCouldNotBeCompleted() const override {
+    // Hack! When the compiler ends abnormally, but there are no contents in
```

Comment:
Mm, doesn't seem to me like a hack; this is the intended and to-be-documented behaviour / encoding of "incomplete compilation of batch member in a batch with an erroneous primary", no?

Reply:
I am happy to remove "Hack!" As you point out, it is the discarding of warnings, etc., that led me to think this is a hack. I will elaborate the comment.